### PR TITLE
A lot of places: Add a CMake port

### LINKED ISSUE #90: LibC: Rewrite scanf() family of functions
The scanf() family of functions in LibC/scanf.cpp should be rewritten, since they are borrowed from an outside source.
The current version doesn't always work correctly, so there's good reason beyond just NIH.

### PR DESCRIPTION
**Pull request by alimpfard**
Current status:
- Builds, and works (can configure Serenity!)
- Takes about ~two hours~ 80m to build
- ~Needs to NIH `scanf` (#90)~
- Probably needs code cleanup

Todos before ready to review:
- [x] Fix #90
- [x] Cleanup some messy changes (the "bump image size" commit)
- [x] Click that "Ready for review" button

### CODE REVIEW

File: Userland/Libraries/LibC/sys/types.h [hunk 1]
```diff
@@ -99,7 +99,7 @@ typedef struct __pthread_cond_t {
 } pthread_cond_t;
 
 typedef void* pthread_rwlock_t;
-typedef void* pthread_rwlockatrr_t;
+typedef void* pthread_rwlockattr_t;
```
Comment (alimpfard) [inline Userland/Libraries/LibC/sys/types.h:102]: @awesomekling :laughing:

File: Ports/cmake/package.sh [hunk 2]
```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env -S bash ../.port_include.sh
+port=cmake
+version=3.19.4
+useconfigure=false
+files="https://github.com/Kitware/CMake/releases/download/v$version/cmake-$version.tar.gz cmake-$version.tar.gz"
+depends=""
```
Comment (linusg) [inline Ports/cmake/package.sh:6]: Please add to AvailablePorts.md :)

Comment (alimpfard) [reply Ports/cmake/package.sh:6]: You keep telling me this, and I keep forgetting about it ^^'

Comment (linusg) [reply Ports/cmake/package.sh:6]: Linting of available ports table when

Comment (alimpfard) [reply Ports/cmake/package.sh:6]: @BenWiederhake ^

Comment (BenWiederhake) [reply Ports/cmake/package.sh:6]: > Linting of available ports table when

https://www.nieuweplaat.nl/wp-content/uploads/2015/11/12045799_892741494130158_3452631501732855933_o.jpg

File: Userland/Libraries/LibPthread/pthread.cpp [hunk 3]
```diff
@@ -685,47 +685,229 @@ int pthread_equal(pthread_t t1, pthread_t t2)
     return t1 == t2;
 }
 
+// FIXME: Use the fancy futex mechanism above to write an rw lock.
+//        For the time being, let's just use a less-than-good lock to get things working.
```
Comment (alimpfard) [inline Userland/Libraries/LibPthread/pthread.cpp:689]: cc @tomuta, @bugaevc - please don't kill me for the many millions of weird bugs here, it's my first time writing an rwlock, much less one that that conforms to the `pthread` APIs :P

File: Ports/cmake/package.sh [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+port_path=$(realpath $(dirname ${BASH_SOURCE[0]}))
+
+build() {
+    return
+}
+
+install() {
+    cmake_dir="${SERENITY_ROOT}/Build/Root/home/anon/Source/cmake"
+    rm -rf "$cmake_dir"
+    mkdir -p "$cmake_dir"
```
Comment (linusg) [inline Ports/cmake/package.sh:17]: I don't see harm in using `run` here as well, it `cd`'s needlessly but at least that makes it print the current command to see what's going on.

```suggestion
    run rm -rf "$cmake_dir"
    run mkdir -p "$cmake_dir"
```

File: Ports/cmake/patches/README.md [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+
+## `0028-cmake-disable-tests.patch`
+
+We don't care about building tests for now, and it makes the compilation much faster.
+
+### Status
+- [ ] Local?
+- [ ] Should be merged to upstream?
+- [ ] Resolves issue(s) with our side of things
+- [ ] Hack
```
Comment (linusg) [inline Ports/cmake/patches/README.md:244]: :100: for adding this file!

File: Ports/cmake/package.sh (hunk 4 above)
Comment (alimpfard) [reply Ports/cmake/package.sh:17]: Sure, why not :shrug:

File: Ports/cmake/patches/README.md (hunk 5 above)
Comment (alimpfard) [reply Ports/cmake/patches/README.md:244]: It's a great idea to explain the patches, especially when they're non-trivial

File: Ports/cmake/patches/README.md [hunk 6]
```diff
@@ -0,0 +1,244 @@
...
+
+These two defines make GCC very sad. reasons are unknown at this time.
+
+### Status
+- [ ] Local?
+- [ ] Should be merged to upstream? 
+- [X] Resolves issue(s) with our side of things
+- [X] Hack
+
+## `0018-isfinite.patch``
```
Comment (linusg) [inline Ports/cmake/patches/README.md:144]:
```suggestion
## `0018-isfinite.patch`
```

File: Userland/Libraries/LibC/sys/types.h (hunk 1 above)
Comment (sunverwerth) [reply Userland/Libraries/LibC/sys/types.h:102]: rwlock go trrrrrr

**Pull request merged**